kernel/src/scan/mod.rs

```diff
@@ -95,7 +95,7 @@ impl ScanBuilder {
         let (all_fields, read_fields, have_partition_cols) = get_state_info(
             logical_schema.as_ref(),
             &self.snapshot.metadata().partition_columns,
-            self.snapshot.column_mapping_mode,
+            self.snapshot.table_properties().get_column_mapping_mode(),
```

Comment:
We don't normally use `get_` for field accessors?
```suggestion
            self.snapshot.table_properties().column_mapping_mode(),
```
(rust can handle structs with both `foo: Foo` and `fn foo(&self) -> &Foo` -- even if both are public)

Reply:
I'm curious to hear everyone's thoughts on this.. I think we need two things here (and the API above was attempting to answer the second part)
1. a struct to parse the table properties into (`TableProperties` struct) which lets users (or engines) examine the table properties that are set
2. an API for checking table properties/feature enablement. This likely needs to check if table features are enabled in the protocol and check table properties being set and perhaps check on other dependent features. (i.e. we need somewhere that we can embed this logic)

this probably isn't the PR for tackling (2) - but would love to hear some thoughts so we could get started on a follow-up. For now I'll just do `column_mapping_mode()` to unblock this and we can iterate?

Reply:
cc @nicklan @scovich

Reply:
I think it makes a lot of sense. These are checks that are going to be made for every table feature, so it helps to have all of it in one place.

Reply:
just gonna take it as a follow up https://github.com/delta-io/delta-kernel-rs/issues/508

and as @nicklan suggested: probably just wait a little longer until we have more use cases and then can inform how we want to build something that unifies protocol/table properties